### PR TITLE
Cross ent with logits

### PR DESCRIPTION
**Pull request by timsetsfire**
main add - CrossEntropyWithLogits.  This is more numerically stable than the original CrossEntropy since the fix for the gradient issue.  

ancillary add - Identity activation function

misc add - MSE loss function, SoftMaxWithLogits loss function.

### CODE REVIEW

**Comment by timsetsfire** [comment]
need to close - i found an issue.
